cdap-ui/app/cdap/components/RulesEngineHome/RulesTab/Rule/index.js

```diff
@@ -0,0 +1,259 @@
+    isDragging: monitor.isDragging()
+  };
+}
+class Rule extends Component {
+  static propTypes = {
+    rule: PropTypes.object,
+    connectDragSource: PropTypes.func.isRequired,
+    isDragging: PropTypes.bool.isRequired
+  };
+
+  state = {
+    viewDetails: false,
+    detailsLoading: false,
+    ruleDetails: null,
+    edit: false
+  };
+
+  componentDidMount() {
+    RulesEngineStore.subscribe(() => {
+      let {rules} = RulesEngineStore.getState();
+      if (rules.activeRuleId === this.props.rule.id) {
+        this.fetchRuleDetails();
+      } else {
+        if (this.state.viewDetails) {
+          this.setState({
+            viewDetails: false
+          });
+        }
+      }
+    });
+  }
+
+  viewDetails = () => {
+    if (this.state.viewDetails) {
+      RulesEngineStore.dispatch({
+        type: RULESENGINEACTIONS.RESETACTIVERULE
+      });
+      return;
+    }
+    RulesEngineStore.dispatch({
+      type: RULESENGINEACTIONS.SETACTIVERULE,
+      payload: {
+        activeRuleId: this.props.rule.id
+      }
+    });
+  };
+
+  fetchRuleDetails = () => {
+    if (this.state.viewDetails) {
+      return;
+    }
+    let {selectedNamespace: namespace} = NamespaceStore.getState();
+    this.setState({
+      viewDetails: true,
+      detailsLoading: true
+    });
+    MyRulesEngineApi
+      .getRuleDetails({
+        namespace,
+        ruleid: this.props.rule.id
+      })
+      .subscribe(
+        res => {
+          this.setState({
+            ruleDetails: res.values,
+            detailsLoading: false
+          });
+        },
+        () => {
+          this.setState({
+            detailsLoading: false
+          });
+        }
+      );
+  };
+
+  onRulesChange(rows) {
+    // TODO: add edit feature for rule in rules tab.
+    console.log(rows);
+  }
+
+  onRuleBookSelect = (rulebookid) => {
+    let {selectedNamespace: namespace} = NamespaceStore.getState();
+    MyRulesEngineApi
+      .addRuleToRuleBook({
+        namespace,
+        rulebookid,
+        ruleid: this.props.rule.id
+      })
+      .subscribe(
+        () => {
+          getRuleBooks();
+        },
+        (err) => {
+          RulesEngineStore.dispatch({
+            type: RULESENGINEACTIONS.SETERROR,
+            payload: {
+              error: {
+                showError: true,
+                message: typeof err === 'string' ? err : err.response.message
+              }
+            }
+          });
+        }
+      );
+  }
+
+  renderDetails = () => {
+    if (this.state.detailsLoading) {
+      return (
+        <Col xs="12" className="text-xs-center">
+          <LoadingSVG />
+        </Col>
+      );
+    }
+    let rules = this.state.ruleDetails.action.map(action => ({
+      uniqueId: shortid.generate(),
+      property: action
+    }));
+    return (
+      <Col xs="12">
+        <Form onSubmit={preventPropagation}>
+          <FormGroup row>
+            <Label sm={2}> {T.translate(`commons.when`)} </Label>
```

Comment:
Yes. I am working with Lea on this. Should have both in the subsequent PR where we enable editing the rule.